lib/streamlit/elements/widgets/button.py

```diff
@@ -339,16 +340,28 @@ def _download_button(
         use_container_width: bool = False,
         ctx: Optional[ScriptRunContext] = None,
     ) -> bool:
-
         key = to_key(key)
         check_session_state_rules(default_value=None, key=key, writes_allowed=False)
+
+        id = compute_widget_id(
+            "button",
```

Comment:
nit: should this type be `download_button`?

Also heads up that this change may cause some weird behavioral issues with #7056 that may not show up as merge conflicts.